docs/API_DESIGN.md

```diff
@@ -89,23 +93,30 @@ CLASS Jwk
   PUBLIC DATA d: string?
 ```
 
-## `InMemoryKeyManager`
+## Key Managers 
+
+### `KeyManager`
+
+```pseudocode!
+INTERFACE KeyManager
+  /// Returns the signer for the given public key.
+  METHOD get_signer(public_jwk: Jwk): Signer
+```
+
+### `InMemoryKeyManager`
 
 ```pseudocode!
 /// An encapsulation of key material stored in-memory.
-CLASS InMemoryKeyManager
-  /// Generates new key material and returns the public key represented as a Jwk.
-  METHOD generate_key_material(): Jwk
+CLASS InMemoryKeyManager IMPLEMENTS KeyManager
+  CONSTRUCTOR(private_jwks: []Jwk)
 
-  /// Returns the Ed25519Signer for the given public key.
-  METHOD get_signer(public_key: Jwk): Ed25519Signer
+  /// Returns the signer for the given public key.
+  METHOD get_signer(public_jwk: Jwk): Signer
 
   /// For importing keys which may be stored somewhere such as environment variables. Return Jwk is the public key for the given private key.
-  METHOD import_key(private_key: Jwk): Jwk
+  METHOD import_private_jwk(private_jwk: Jwk): Jwk
```

Comment:
Yea its not 100% clear from a new users prospective  (unless they read the documentation) that the return JWK will be public instead of private.

So this seems like another datapoint to have a light wrapper around Jwk for PrivateJwk and PublicJwk

Reply:
is that perhaps the reason why the current implementation would return keyId string instead of the public key, after storing the private key?